README.md

```diff
@@ -1,19 +1,49 @@
 # Brooklyn CLI
 
-## Compiling
-
-1. Ensure your [$GOPATH](http://golang.org/cmd/go/#hdr-GOPATH_environment_variable) is set correctly,
-   to some location where Go does its work, such as `~/gocode` .
-2. Get and build the cli source code: `go get github.com/brooklyncentral/brooklyn-cli/br`
-3. Run it from `$GOPATH/bin/br` (add `$GOPATH/bin/` to your PATH).
-4. Thereafter if you want to do code changes, 
-   link the `$GOPATH/src/github.com/brooklyncentral/brooklyn-cli`
-   with the directory where you want to keep your git repositories.
-   (TODO: clarify best practice for this, including how to combine
-   it with a Brooklyn all-projects build)
+## Toolchain
+
+You will need the following tools to build the CLI:
+- Go (min version 1.5.1), with full cross-compiler support (see https://golang.org/doc/install/source).
+  On Mac, if using Homebrew, use "brew install go --with-cc-all"
+- godep (see https://github.com/tools/godep)
+
+Optional:
+- Maven (used by the Brooklyn build process)
+
+
+## Build Pre-Requisites
+
+- Ensure your [$GOPATH](http://golang.org/cmd/go/#hdr-GOPATH_environment_variable) is set correctly 
+  to a suitable location for your Go code.
+- git clone the CLI code into $GOPATH/src/github.com/brooklyncentral/brooklyn-cli
+
+## Compiling the code with Go for development purposes
+
```

Comment:
This section should maybe detail on using the 'external' dependencies rather than those cached with godep.  Would need to mention that if a compile error occurs then to try using the godep cached dependencies; also potential for using `go clean -i` and/or `godep go clean -i` to remove old object/archive files.